src/GenerateB.c

```diff
@@ -181,10 +177,89 @@ void InitBeam(void)
 					default: LogError(ONE_POS,"Incompatibility error in GenerateB");
 				}
 				beam_descr=dyn_sprintf("Gaussian beam (%s)\n"
-				                       "\tWidth="GFORMDEF" (confinement factor s="GFORMDEF")\n"
-				                       "\tCenter position: "GFORMDEF3V,tmp_str,w0,s,COMP3V(beam_center_0));
+				                       "\tWidth="GFORMDEF" (confinement factor s="GFORMDEF")\n",tmp_str,w0,s);
+			}
+			return;
+#ifndef NO_FORTRAN
+		case B_BES_CS:
+		case B_BES_CSp:
+		case B_BES_M:
+		case B_BES_LE:
+		case B_BES_LM:
+		case B_BES_TEL:
+		case B_BES_TML:
+			if (surface) PrintError("Currently, Bessel incident beam is not supported for '-surf'");
+			// initialize parameters
+			n0=round(beam_pars[0]);
+			alpha0=beam_pars[1];
```

Comment:
The input angle should be in degrees (as other input angles in ADDA - this already described as such in help strings). See function Rad2Deg.
Moreover, we should probably check it to be between 0 and 90 degrees. Angle 0 should not be allowed unless the order is 0 (or some specific matrix M for orders +-1 or +-2 - I am not sure, please look carefully at the formulae). Similar issue is for angle 90.